payments/models.py

```diff
@@ -775,12 +775,13 @@ def update_lease_and_application(self, new_status):
 
         self.lease.save(update_fields=["status", "comment"])
 
-        if new_status in OrderStatus.get_paid_statuses():
-            if self.lease.application:
-                application = self.lease.application
+        if application := self.lease.application:
+            if new_status == OrderStatus.REJECTED:
+                application.status = ApplicationStatus.REJECTED
+                application.save(update_fields=["status"])
+            elif new_status in OrderStatus.get_paid_statuses():
                 application.status = ApplicationStatus.HANDLED
-                self.lease.application.save(update_fields=["status"])
-
+                application.save(update_fields=["status"])
```

Comment:
At this point, would it make sense to extract the application logic to a util function like `get_lease_status`?

Reply:
added get_application_status. I also moved the logic for generating sticker numbers to its own function.